app-config-extensions/src/index.test.ts

```diff
@@ -843,6 +843,31 @@ describe('$env directive', () => {
     const parsed2 = await source.read([envDirective()]);
     expect(parsed2.toJSON()).toEqual({ sibling: true, testing: false });
   });
+
+  it('overrides env', async () => {
+    await withTempFiles(
+      {
+        'test-file.yml': `
+          foo:
+            $env:
+              default: 44
+              dev: 88
+        `,
+      },
+      async (inDir) => {
+        const source = new LiteralSource({
+          $extends: {
+            path: inDir('test-file.yml'),
+            env: 'development',
+          },
```

Comment:
New feature - `$extends` with a `env` option that parses the referenced tree using that currentEnvironment value